### DISCUSSION TITLE
Terraform Connector: Error - Primary key for stream: {xxx} is already pre-defined

### DISCUSSION THREAD

**Discussion by jduckwo** [airbytehq/airbyte#40604] [Connector Questions]
I am doing a POC with Terraform. I have a connection from one Postgres database to another Postgres database. The connection has one stream in it. When I apply the stream via terraform the connection is created and I am able to run a job for it successfully. When I try to add a second stream to the config and apply the terraform, I get the following error:

> Error: failure to invoke API
> 
> with airbyte_connection.connection__source_1_db--central_db, on connections.tf line 6, in resource "airbyte_connection" "connection__source_1_db--central_db": 6: resource "airbyte_connection" "connection__source_1_db--central_db" {
> 
> unknown status code returned: Status 400
> {
> "detail":"The body of the request contains an invalid connection configuration. Primary key for stream: table_1 is already pre-defined. Please do NOT include a primary key configuration for this stream.","type":"https://reference.airbyte.com/reference/errors","title":"bad-request","status":400
> }

We should be able to add new streams over time without having to destroy and reapply the config. Are we doing something wrong or is this in fact a bug with the Terraform provider? This is a pretty big deal and I am sure it is a blocker for many people if this feature doesn't work.

### Initial Config: 1st `terraform apply`

```
resource "airbyte_connection" "connection__source_1_db--central_db" {
  name = "source_1_db -> central_db"

  source_id      = airbyte_source_postgres.source__source_1_db.source_id
  destination_id = airbyte_destination_postgres.destination__central_db.destination_id

  non_breaking_schema_updates_behavior = "propagate_fully"
  schedule = {
    schedule_type   = "manual"
  }
  status = "active"

  configurations = {
      streams = [
      {
        name = "table_1"
        sync_mode = "incremental_deduped_history"
      }
    ]
  }

}
```

### Adding Additional Stream: 2nd `terraform apply`

```
resource "airbyte_connection" "connection__source_1_db--central_db" {
  name = "source_1_db -> central_db"

  source_id      = airbyte_source_postgres.source__source_1_db.source_id
  destination_id = airbyte_destination_postgres.destination__central_db.destination_id

  non_breaking_schema_updates_behavior = "propagate_fully"
  schedule = {
    schedule_type   = "manual"
  }
  status = "active"

  configurations = {
      streams = [
      {
        name = "table_1"
        sync_mode = "incremental_deduped_history"
      },
      {
        name = "table_2"
        sync_mode = "incremental_deduped_history"
      }    ]
  }

}
```

**Comment by makz81** [comment]
I experienced the same error a lot and it's pretty annoying. However, I found out that it becomes problematic when you change the order of the streams. It tries to apply the same config to the stream again, which is problematic. Only the new stream should appear in the apply.

(Also, you have two streams with the same name here, but I think that's just due to the example, right?)

**Reply by jduckwo** [reply]
Yeah, that table name was just for the example. I fixed the original post above to avoid future confusion.

Were you ever able to find a fix for this issue?

[a comment by an automated account is not shown]
